app/services/bulk_upload/lettings/year2024/row_parser.rb

```diff
@@ -584,19 +584,23 @@ def assigned_to
   end
 
   def validate_uprn_exists_if_any_key_address_fields_are_blank
-    if field_16.blank? && (field_17.blank? || field_19.blank?)
+    if field_16.blank? && !key_address_fields_provided?
       errors.add(:field_16, I18n.t("validations.not_answered", question: "UPRN"))
     end
   end
 
   def validate_address_option_found
-    if log.uprn_selection.nil? && field_16.blank? && (field_17.present? || field_19.present?)
+    if log.uprn.nil? && field_16.blank? && key_address_fields_provided?
       %i[field_17 field_18 field_19 field_20 field_21 field_22].each do |field|
         errors.add(field, I18n.t("validations.no_address_found"))
       end
     end
   end
 
+  def key_address_fields_provided?
+    field_17.present? && field_19.present? && field_21.present? && field_22.present?
+  end
```

Comment:
Postcode (field_21 and field_22) are optional (cause we could use the LA instead), are we sure we want to validate it like this? What values would need to be in the BU file to upload a successful, completed log without providing a postcode or UPRN?